backend/api/Controllers/RobotController.cs

```diff
@@ -709,16 +709,24 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
 
         mission.UpdateWithIsarInfo(isarMission);
         mission.Status = MissionStatus.Ongoing;
-
         await _missionService.Create(mission);
 
         robot.Status = RobotStatus.Busy;
         robot.CurrentMissionId = mission.Id;
+
+        var deck = await _assetDeckService.ReadById(scheduleLocalizationMissionQuery.DeckId);
+
+        if (deck == null)
+        {
+            _logger.LogWarning("Could not find deck with id={id}", scheduleLocalizationMissionQuery.DeckId);
+            return NotFound("Deck not found");
+        }
```

Comment:
Consider moving this before running
"isarMission = await _isarService.StartLocalizationMission(robot, scheduleLocalizationMissionQuery.LocalizationPose);"
such that we cancel the localization if we cant find the correct deck

Reply:
Done !